nixos/modules/services/networking/netmaker/netmaker.nix

```
@@ -0,0 +1,787 @@
+      description = lib.mdDoc ''
+        The domain Netmaker Server's API is exposed on.
+      '';
+    };
+    ui.domain = lib.mkOption {
+      type = with lib.types; str;
+      default = "dashboard.${config.services.netmaker.domain}";
+      defaultText = lib.literalExpression ''"dashboard.''${config.services.netmaker.domain}"'';
+      description = lib.mdDoc ''
+        The domain Netmaker UI is exposed on.
+      '';
+    };
+    mqtt.domain = lib.mkOption {
+      type = with lib.types; str;
+      default = "broker.${cfg.domain}";
+      defaultText = lib.literalExpression ''"broker.''${config.services.netmaker.domain}"'';
+      description = lib.mdDoc ''
+        Domain the message queue is proxied through for external access.
+      '';
+    };
+
+    package = lib.mkOption {
+      type = with lib.types; package;
+      default = if cfg.installType == "ce" then pkgs.netmaker else pkgs.netmaker-pro;
+      defaultText = lib.literalExpression ''if cfg.installType == "ce" then pkgs.netmaker else pkgs.netmaker-pro'';
+      description = lib.mdDoc ''
+        Package providing `netmaker` binary.
+      '';
+    };
+
+    nmctl.package = lib.mkOption {
+      type = with lib.types; package;
+      default = pkgs.nmctl;
+      defaultText = lib.literalExpression ''pkgs.nmctl'';
+      description = lib.mdDoc ''
+        Package providing `nmctl` binary.
+      '';
+    };
+
+    internalListenAddress = lib.mkOption {
+      type = with lib.types; str;
+      default = "127.0.0.2";
+      description = lib.mdDoc ''
+        An address services will be listening on before being proxied
+      '';
+    };
+
+    verbosity = lib.mkOption {
+      type = with lib.types; enum [ 1 2 3 4 ];
+      default = 1;
+      description = lib.mdDoc ''
+        Verbosity level of Netmaker Server logging.
+      '';
+    };
+
+    telemetry = lib.mkEnableOption "sending telemetry to Netmaker developers";
+
+    environment = lib.mkOption {
+      type = with lib.types; attrsOf str;
+      default = { };
+      apply = envAttrs: pkgs.writeText "environment-env" (lib.trivial.pipe envAttrs [
+        (lib.attrsets.mapAttrsToList (name: value: "${name}=${lib.strings.escapeShellArg value}"))
+        (builtins.concatStringsSep "\n")
+      ]);
+      description = lib.mdDoc ''
+        Additional environment variables to set on Netmaker Server
+      '';
+    };
+
+    environmentFiles = lib.mkOption {
+      type = with lib.types; listOf path;
+      default = [ ];
+      description = lib.mdDoc ''
+        Additional files holding environment variable definitions to set on Netmaker Server
+      '';
+    };
+
+    config = lib.mkOption {
+      type = with lib.types; anything;
+      description = lib.mdDoc ''
+        Content of Netmaker configuration file.
+
+        see for more information:
+        - https://github.com/gravitl/netmaker/blob/dc8f9b1bc74d262669bf96fab5f0e545f5906432/config/config.go
```

```suggestion
        <https://github.com/gravitl/netmaker/blob/dc8f9b1bc74d262669bf96fab5f0e545f5906432/config/config.go>
```
should we code the version of the package into the url instead?